File: Sample/2.0/Models/Sample/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyApp.Models.Sample
{
  [Table("Products", Schema = "dbo")]
  public class Product
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id
    {
      get;
      set;
    }


    [InverseProperty("Product")]
    public ICollection<OrderDetail> OrderDetails { get; set; }
    public string ProductName
    {
      get;
      set;
    }
    public string ProductPicture
    {
      get;
      set;
    }
    public decimal ProductPrice
    {
      get;
      set;
    }
  }
}

File: Sample/2.0_VB/Models/Product.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace MyApp.Models.Sample
    <Table("Products", Schema:="dbo")>
    Public Class Product
        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Id() As Integer

        <InverseProperty("Product")>
        Public Property OrderDetails() As ICollection(Of OrderDetail)

        Public Property ProductName() As String

        Public Property ProductPicture() As String

        Public Property ProductPrice() As Decimal

    End Class
End Namespace
